File: Models/CalculatorState.cs


```csharp
using System;

namespace Models
{
    public class CalculatorState
    {
        public double FirstNumber { get; set; }
        public double SecondNumber { get; set; }
        public string CurrentState { get; set; }
        public string DecimalFormat { get; set; }
        public string CurrentEntry { get; set; }

        public CalculatorState()
        {
            FirstNumber = 0;
            SecondNumber = 0;
            CurrentState = "initial";
            DecimalFormat = "0.00";
            CurrentEntry = "";
        }
    }
}
```

File: frontend/reducers/CalculatorState.js


import { createStore, combineReducers } from 'redux';

const initialState = {
  firstNumber: 0,
  secondNumber: 0,
  currentState: 'initial',
  decimalFormat: false,
  currentEntry: ''
};

const resetCalculatorState = () => {
  return { type: 'RESET_CALCULATOR_STATE' };
};

const calculatorReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'RESET_CALCULATOR_STATE':
      return initialState;
    default:
      return state;
  }
};

const rootReducer = combineReducers({
  calculator: calculatorReducer
});

const store = createStore(rootReducer);

export { resetCalculatorState, store };